tests/file_uploads/tests.py

```diff
@@ -435,6 +435,22 @@ def test_broken_custom_upload_handler(self):
             with self.assertRaisesMessage(AttributeError, msg):
                 self.client.post('/quota/broken/', {'f': file})
 
+    def test_upload_stopped(self):
+        with tempfile.NamedTemporaryFile() as temp_file:
+            path = temp_file.name
+            self.assertIs(os.path.exists(path), True)
+            response = self.client.post('/upload_stopped/', {'file': temp_file})
+            temp_path = response.json()['temp_path']
+            self.assertIs(os.path.exists(temp_path), False)
```

Comment:
I had to mocked the `Parser` because handling `StopIteration` inside generators had changed in Python 3.7+ (see [PEP-479](https://www.python.org/dev/peps/pep-0479/)).